framework/tlvf/src/src/WSC/AttrList.cpp

```diff
@@ -0,0 +1,504 @@
+        case ATTR_DEV_NAME:
+            if (!addAttr<cWscAttrDeviceName>())
+                return false;
+            break;
+        case ATTR_DEV_PASSWORD_ID:
+            if (!addAttr<cWscAttrDevicePasswordID>())
+                return false;
+            break;
+        case ATTR_ENROLLEE_NONCE:
+            if (!addAttr<cWscAttrEnrolleeNonce>())
+                return false;
+            break;
+        case ATTR_MAC_ADDR:
+            if (!addAttr<cWscAttrMac>())
+                return false;
+            break;
+        case ATTR_MANUFACTURER:
+            if (!addAttr<cWscAttrManufacturer>())
+                return false;
+            break;
+        case ATTR_MSG_TYPE:
+            if (!addAttr<cWscAttrMessageType>())
+                return false;
+            break;
+        case ATTR_MODEL_NAME:
+            if (!addAttr<cWscAttrModelName>())
+                return false;
+            break;
+        case ATTR_MODEL_NUMBER:
+            if (!addAttr<cWscAttrModelNumber>())
+                return false;
+            break;
+        case ATTR_OS_VERSION:
+            if (!addAttr<cWscAttrOsVersion>())
+                return false;
+            break;
+        case ATTR_PUBLIC_KEY:
+            if (!addAttr<cWscAttrPublicKey>())
+                return false;
+            break;
+        case ATTR_RF_BANDS:
+            if (!addAttr<cWscAttrRfBands>())
+                return false;
+            break;
+        case ATTR_SERIAL_NUMBER:
+            if (!addAttr<cWscAttrSerialNumber>())
+                return false;
+            break;
+        case ATTR_WSC_STATE:
+            if (!addAttr<cWscAttrWscState>())
+                return false;
+            break;
+        case ATTR_UUID_E:
+            if (!addAttr<cWscAttrUuidE>())
+                return false;
+            break;
+        case ATTR_VENDOR_EXTENSION:
+            if (!addAttr<cWscVendorExtWfa>())
+                return false;
+            break;
+        case ATTR_VERSION:
+            if (!addAttr<cWscAttrVersion>())
+                return false;
+            break;
+        case ATTR_PRIMARY_DEV_TYPE:
+            if (!addAttr<cWscAttrPrimaryDeviceType>())
+                return false;
+            break;
+        case ATTR_REGISTRAR_NONCE:
+            if (!addAttr<cWscAttrRegistrarNonce>())
+                return false;
+            break;
+        case ATTR_UUID_R:
+            if (!addAttr<cWscAttrUuidR>())
+                return false;
+            break;
+        case ATTR_ENCR_SETTINGS:
+            if (!addAttr<cWscAttrEncryptedSettings>())
+                return false;
+            break;
+        case ATTR_AUTHENTICATOR:
+            if (!addAttr<cWscAttrAuthenticator>())
+                return false;
+            break;
+        // TODO these are not expected attributes (part of config data)
+        case ATTR_SSID:
+        case ATTR_NETWORK_KEY:
+        case ATTR_KEY_WRAP_AUTH:
+        case ATTR_ENCR_TYPE:
+        case ATTR_AUTH_TYPE:
```

Comment:
Should add our `FALLTHROUGH` macro. Otherwise it can generate build errors on some compilers.